checkstyle/README.adoc

```diff
@@ -0,0 +1,102 @@
+= checkstyle plugin
+
+== Usage
+
+Enforces certain coding standards, complements both `enforcer` and `formatter` plugins.
+
+Focuses on best practices for code and not stylistic formatting as that is done by `formatter` plugin.
+
+Checks are automatically done once included to project.
+
+You can manually test checkstyle rules with `mvn checkstyle:check`
+
+Documentation and further configuration for each check can be found https://checkstyle.org/checks.html[in their documentation]
+
+=== Add the following to pom.xml
+
+.pom.xml
+[source,xml]
+----
+<plugin>
+  <groupId>org.apache.maven.plugins</groupId>
+  <artifactId>maven-checkstyle-plugin</artifactId>
+  <version>3.5.0</version>
+  <configuration>
+    <logViolationsToConsole>true</logViolationsToConsole>
+    <checkstyleRules>
+      <module name="Checker">
```

Comment:
add Naming Conventions
PackageName ^com.teragrep.[a-z]{3}_\d{2}(\.[a-zA-Z_]\w*)*$
TypeName

Reply:
Added but with `^com\.teragrep\.[a-z]{3}_\d{2}(.[a-zA-Z]\w*)*$` naming so `com.teragrep.pth_10` matches, also escaped the dots